torch/csrc/jit/ir/alias_analysis.cpp

```diff
@@ -1127,15 +1127,6 @@ void AliasDb::makePointerTo(const Value* from, const Value* to) {
   // immutable. `Any` is mutable but can point to an immutable type
   // through refinement
   if (isMutableTypeInternal(from) != isMutableTypeInternal(to)) {
```

Comment:
didn't look at any of the other code in here, but we should still be making a pointer if there is any type overlap between the sets. please tag me for changes related to alias analysis. this should actually have been changed as part of the Union changes... maybe you could factor out the code from refinement into a "has type overlap" function  or something similar ? it would be nice if we could reuse that logic to peephole optimize away `prim::isinstance`

Reply:
`(isMutableTypeInternal(from) != isMutableTypeInternal(to))` we're checking equality not overlap here

Reply:
Sorry for the wording choice, I did mean equality

Reply:
i think we want to be checking overlap, e.g. Optional[List[int]] should have a pointer to List[int]